deps/chakrashim/src/v8handlescope.cc

```diff
@@ -54,28 +54,38 @@ HandleScope *HandleScope::GetCurrent() {
 }
 
 bool HandleScope::AddLocal(JsValueRef value) {
-  if (_count < _countof(_locals)) {
+  if (_count <= kOnStackLocals) {
     _locals[_count++] = value;
-    return true;
+    if (_count < kOnStackLocals) {
+      return true;
+    }
   }
 
+  // Ran out of _locals. Push everything in array and empty
+  // the _locals array.
   if (_refs == JS_INVALID_REFERENCE) {
     if (JsCreateArray(1, &_refs) != JsNoError) {
       return AddLocalAddRef(value);
     }
   }
 
-  JsValueRef index;
+  JsValueRef saveInHandleScopeFunction =
+    jsrt::ContextShim::GetCurrent()->GetsaveInHandleScopeFunction();
 
-  if (JsIntToNumber(_count - _countof(_locals), &index) != JsNoError) {
-    return AddLocalAddRef(value);
+  JsValueRef args[kOnArgsToPush];
```

Comment:
`kOnArgsToPush` is completely local, maybe declaring it here, or just use `kOnStackLocals + 2` (JsCallFunction can use _countof) is easier to read and understand?